powerauth-java-server/src/main/java/com/wultra/security/powerauth/app/server/service/util/jwt/MLDSASigner.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * PowerAuth Server and related software components
+ * Copyright (C) 2025 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.wultra.security.powerauth.app.server.service.util.jwt;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSSigner;
+import com.nimbusds.jose.crypto.impl.AlgorithmSupportMessage;
+import com.nimbusds.jose.jca.JCAContext;
+import com.nimbusds.jose.util.Base64URL;
+import com.wultra.security.powerauth.crypto.lib.model.exception.GenericCryptoException;
+import com.wultra.security.powerauth.crypto.lib.v4.PqcDsa;
+import lombok.AllArgsConstructor;
+
+import java.security.PrivateKey;
+import java.util.Set;
+
+/**
+ * Implementation of the {@link JWSSigner} to support {@link PqcDsa}.
+ *
+ * @author Jan Pesek, jan.pesek@wultra.com
+ */
+@AllArgsConstructor
+public class MLDSASigner implements JWSSigner {
+    private static final PqcDsa PQC_DSA = new PqcDsa();
+    private static final Set<JWSAlgorithm> SUPPORTED_ALGORITHMS = Set.of(JWSAlgorithmMLDSA.MLDSA65);
+    private final JCAContext jcaContext = new JCAContext();
+
+    private final PrivateKey privateKey;
+
+    @Override
+    public Base64URL sign(final JWSHeader header, final byte[] signingInput) throws JOSEException {
+        final JWSAlgorithm alg = header.getAlgorithm();
+
+        if (!supportedJWSAlgorithms().contains(alg)) {
+            throw new JOSEException(AlgorithmSupportMessage.unsupportedJWSAlgorithm(alg, this.supportedJWSAlgorithms()));
+        }
+
+        try {
+            final byte[] signature = PQC_DSA.sign(privateKey, signingInput);
+            return Base64URL.encode(signature);
+        } catch (GenericCryptoException e) {
```

Comment:
We should either log this exception or propagate the cause. The cause seems to be lost.